src/service.ts

```diff
@@ -2,40 +2,66 @@ process.env.FORCE_COLOR = "0";
 
 import LRU from "nanolru";
 import path from "path";
-import prettier from "prettier";
-import { Options, format } from "prettier";
+import type { Options } from "prettier";
 
 const configCache = new LRU({ max: 20, maxAge: 60000 });
 
-async function resolveConfig(cwd: string, filepath: string): Promise<Options> {
+async function resolveConfig(
+  prettier: any,
+  cwd: string,
+  filepath: string
+): Promise<Options> {
   let v = configCache.get<string, Options>(cwd);
+
   if (!v) {
     v = await prettier.resolveConfig(filepath, {
       editorconfig: true,
       useCache: false,
     });
+
+    if (!v && typeof process.env.PRETTIERD_DEFAULT_CONFIG === "string") {
+      console.log(process.env.PRETTIERD_DEFAULT_CONFIG)
+      v = await prettier.resolveConfig(null, {
+        config: process.env.PRETTIERD_DEFAULT_CONFIG,
+        editorconfig: false,
+        useCache: false,
+      });
+    }
+
     if (v) {
       configCache.set(cwd, v);
     }
   }
+
   return {
     ...v,
     filepath,
   };
 }
 
+function resolvePrettier(cwd: string): Promise<any> {
```

Comment:
If we introduce the interface, we can use it here as the return type (with a typecast in the function body).